jaeger-thrift/src/main/java/io/jaegertracing/thrift/internal/senders/HttpSender.java

```diff
@@ -69,18 +69,21 @@ public void send(Process process, List<Span> spans) throws SenderException {
       throw new SenderException(String.format("Could not send %d spans", spans.size()), e, spans.size());
     }
 
-    if (!response.isSuccessful()) {
-      String responseBody;
-      try {
-        responseBody = response.body() != null ? response.body().string() : "null";
-      } catch (IOException e) {
-        responseBody = "unable to read response";
-      }
+    if (response.isSuccessful()) {
+      response.close();
+      return;
+    }
 
-      String exceptionMessage = String.format("Could not send %d spans, response %d: %s",
-          spans.size(), response.code(), responseBody);
-      throw new SenderException(exceptionMessage, null, spans.size());
+    String responseBody;
+    try {
+      responseBody = response.body() != null ? response.body().string() : "null";
+    } catch (IOException e) {
+      responseBody = "unable to read response";
     }
+
+    String exceptionMessage = String.format("Could not send %d spans, response %d: %s",
+        spans.size(), response.code(), responseBody);
+    throw new SenderException(exceptionMessage, null, spans.size());
```

Comment:
Don't you need to close, in case the response isn't successful? Is there an alternative where try-with-resources is used, so that our code wouldn't need to care about the closing of the resources?

Reply:
Calling `response.body().string()` will close the response, so this happens fine in the non-successful code path.

Ideally we'd use a try-with-resources, but we can't use it because this project requires Java 6 compatibility.

Reply:
Can't wait to get rid of Java 6 support... (#511)